src/test/addrman_tests.cpp

```diff
@@ -398,8 +405,8 @@ BOOST_AUTO_TEST_CASE(addrman_getaddr)
     // Test 25: Ensure GetAddr still returns 23% when addrman has many addrs.
     for (unsigned int i = 1; i < (8 * 256); i++) {
         int octet1 = i % 256;
```

Comment:
This code isn't at all doing what I expected. On first looking, I assumed it was slicing off 8 bits at a time in order to format a 32 bit int as an ipv4 address, ie I was expecting this:

```
int octet1 = i % 256;
int octet2 = i >> 8 % 256;
int octet3 = i >> 16 % 256;
```

but in fact it's effectively doing this:

```
int octet1 = i % 256;
int octet2 = i >> 8 % 256;
int octet3 = i >> 9 % 256;
```
ie the bits used for octet 2 and 3 are overlapping. Did you work out whether there was a reason for this?

I'd prefer to keep these using the modulos to indicate that each octet should be in the range 0-255. The only reason your suggested change would work is because i never exceeds 2^16. Perhaps replace with the following if there's no good reason to keep the strange octet3 behaviour:

```
int octet1 = i % 256;
int octet2 = i >> 8 % 256;
std::string strAddr = boost::to_string(octet1) + "." + boost::to_string(octet2) + ".0.23";
```
ie eliminate octet3 entirely.